### PR TITLE
Add deployment error reference for ece-tools

### PR DESCRIPTION
**Pull request by meker12**
Merges changes from #7261

## Purpose of this pull request

Adds a deployment error reference for ece-tools errors to the Troubleshooting section of the _Cloud Guide_.

- [x] Technical review
- [ ] Editorial review
- [x] Staging build

## Affected DevDocs pages

- https://devdocs.magento.com/cloud/trouble/error-codes.html

## Links to Magento source code

https://github.com/magento/ece-tools/pull/726

whatsnew
Added an [Error message reference for ece-tools]( https://devdocs.magento.com/cloud/trouble/error-codes.html) to list errors and suggested actions for resolving errors that occur during the Magento Cloud deployment process.

### CODE REVIEW

File: src/cloud/trouble/error-codes.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+| 118 |  | Required utility wasn't found (timeout, bash) | {{site.data.cloud-error-messages.UTILITY_NOT_FOUND}} |
+| 119 | install-update: deploy-static-content | Command `/bin/magento setup:static-content:deploy` failed | {{site.data.cloud-error-messages.CLOUD_LOG_VERBOSE_ACTION}} |
+| 120 | compress-static-content | Static content compression failed | {{site.data.cloud-error-messages.CHECK_CLOUD_LOG_ACTION}} |
+| 121 | deploy-static-content:generate | Cannot update the deployed version | {{site.data.cloud-error-messages.SCD_UNABLE_UPDATE_VERSION}} |
+| 122 | clean-static-content | Failed to clean static content files | |
+| 123 | install-update: split-db | Command `/bin/magento setup:db-schema:split` failed | {{site.data.cloud-error-messages.CLOUD_LOG_VERBOSE_ACTION}} |
+| 124 | clean-view-preprocessed | Failed to clean the `var/view_preprocessed` folder | {{site.data.cloud-error-messages.VIEW_PREPROCESSED_CLEAN_FAILED}} |
+| 125 | install-update: reset-password | Failed to update the `/var/credentials_email.txt` file | {{site.data.cloud-error-messages.FILE_CREDENTIALS_EMAIL_NOT_WRITABLE}} |
+| 126 | install-update: update | Command `/bin/magento setup:upgrade` failed | |
```
Comment (oshmyheliuk) [inline src/cloud/trouble/error-codes.md:77]:
```suggestion
| 126 | install-update: update | Command `/bin/magento setup:upgrade` failed | {{site.data.cloud-error-messages.INSTALL_UPGRADE_ACTION}} |
```

File: src/cloud/trouble/error-codes.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+| 6 | | Unable to read the `.schema.yaml` file | |
+| 7 | refresh-modules | Cannot write to the `./app/etc/config.php` file |{{site.data.cloud-error-messages.CONFIG_PHP_IS_NOT_WRITABLE}} |
+| 8 | validate-config | Cannot read the `composer.json` file |{{site.data.cloud-error-messages.CANT_READ_COMPOSER_JSON}} |
+| 9 | validate-config | Composer.json is missing required autoload section | {{site.data.cloud-error-messages.COMPOSER_MISSED_REQUIRED_AUTOLOAD}} |
+| 10 | validate-config | The file `.magento.env.yaml` contains an option which isn't declared in the schema, or has an invalid value or stage | {{site.data.cloud-error-messages.WRONG_CONFIGURATION_MAGENTO_ENV_YAML}}|
+| 11 | refresh-modules | Command `/bin/magento module:enable --all` failed | {{site.data.cloud-error-messages.CLOUD_LOG_VERBOSE_ACTION}} |
+| 12 | apply-patches | Failed to apply patch | |
+| 13 | set-report-dir-nesting-level | Cannot write to the file `/pub/errors/local.xml` | |
+| 14 | copy-sample-data | Failed to copy sample data files | |
+| 15 | compile-di | Command `/bin/magento setup:di:compile` failed | |
```
Comment (oshmyheliuk) [inline src/cloud/trouble/error-codes.md:34]:
```suggestion
| 15 | compile-di | Command `/bin/magento setup:di:compile` failed | {{site.data.cloud-error-messages.CLOUD_LOG_VERBOSE_ACTION}} |
```

File: src/cloud/trouble/error-codes.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+| 24 | backup-data: static-content | Failed to clean the `./init/pub/static/` directory | {{ site.data.cloud-error-messages.CLEAN_INIT_PUB_STATIC_FAILED}} |
+| 25 | | Cannot find the Composer package | {{site.data.cloud-error-messages.COMPOSER_PACKAGE_NOT_FOUND}} |
+
+## Deploy stage
+
+| Error code | Deploy step | Error description | Suggested action |
+| --- | --- | --- | -- |
+| 101 | pre-deploy: cache | Incorrect cache configuration (missing port or host) | {{site.data.cloud-error-messages.WRONG_CACHE_CONFIGURATION}} |
+| 102 | | Cannot write to the `./app/etc/env.php` file | {{site.data.cloud-error-messages.ENV_PHP_IS_NOT_WRITABLE}} |
+| 103 | | Configuration isn't defined in the `schema.yaml` file  | |
```
Comment (oshmyheliuk) [inline src/cloud/trouble/error-codes.md:52]:
```suggestion
| 103 | | Configuration isn't defined in the `schema.yaml` file  | {{site.data.cloud-error-messages.CONFIG_NOT_DEFINED}} |
```

File: src/cloud/trouble/error-codes.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+| 129 | install-update: reset-password | Enable to read reset password template | |
+
+## Post-deploy stage
+
+| Error code | Post-deploy step | Error description | Suggested action |
+| --- | --- | --- | -- |
+| 201 | is-deploy-failed | Deploy stage failed | |
+| 202 |  | The `./app/etc/env.php` file is not writable | {{site.data.cloud-error-messages.ENV_PHP_IS_NOT_WRITABLE}} |
+| 203 |  | Configuration isn't defined in the `schema.yaml` file | {{site.data.cloud-error-messages.CONFIG_NOT_DEFINED}} |
+| 204 |  | Failed to parse the `.magento.env.yaml` file | |
```
Comment (oshmyheliuk) [inline src/cloud/trouble/error-codes.md:89]:
```suggestion
| 204 |  | Failed to parse the `.magento.env.yaml` file | {{site.data.cloud-error-messages.CONFIG_PARSE_FAILED}} |
```

File: src/cloud/trouble/error-codes.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+| 109 | validate-config | Incorrect database configuration | {{site.data.cloud-error-messages.WRONG_CONFIGURATION_DB}} |
+| 110 | validate-config | Incorrect session configuration | {{site.data.cloud-error-messages.WRONG_CONFIGURATION_SESSION}} |
+| 111 | validate-config | Incorrect search configuration | {{site.data.cloud-error-messages.WRONG_CONFIGURATION_SEARCH}} |
+| 112 | validate-config | Incorrect resource configuration | {{site.data.cloud-error-messages.WRONG_CONFIGURATION_RESOURCE}} |
+| 113 | validate-config:elasticsuite-integrity | ElasticSuite is installed, but the ElasticSearch service is not available | {{site.data.cloud-error-messages.ELASTIC_SUITE_WITHOUT_ES}} |
+| 114 | validate-config:elasticsuite-integrity | ElasticSuite is installed, but another search engine is used | {{site.data.cloud-error-messages.ELASTIC_SUITE_WRONG_ENGINE}} |
+| 115 |  | Database query execution failed | |
+| 116 | install-update: setup | Command `/bin/magento setup:install` failed | {{site.data.cloud-error-messages.INSTALL_UPGRADE_ACTION}} |
+| 117 | install-update: config-import | Command `app:config:import` failed | {{site.data.cloud-error-messages.INSTALL_UPGRADE_ACTION}} |
```
Comment (oshmyheliuk) [inline src/cloud/trouble/error-codes.md:68]:
```suggestion
| 117 | install-update: config-import | Command `app:config:import` failed | {{site.data.cloud-error-messages.CLOUD_LOG_VERBOSE_ACTION}} |
```

File: src/cloud/trouble/error-codes.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+| 120 | compress-static-content | Static content compression failed | {{site.data.cloud-error-messages.CHECK_CLOUD_LOG_ACTION}} |
+| 121 | deploy-static-content:generate | Cannot update the deployed version | {{site.data.cloud-error-messages.SCD_UNABLE_UPDATE_VERSION}} |
+| 122 | clean-static-content | Failed to clean static content files | |
+| 123 | install-update: split-db | Command `/bin/magento setup:db-schema:split` failed | {{site.data.cloud-error-messages.CLOUD_LOG_VERBOSE_ACTION}} |
+| 124 | clean-view-preprocessed | Failed to clean the `var/view_preprocessed` folder | {{site.data.cloud-error-messages.VIEW_PREPROCESSED_CLEAN_FAILED}} |
+| 125 | install-update: reset-password | Failed to update the `/var/credentials_email.txt` file | {{site.data.cloud-error-messages.FILE_CREDENTIALS_EMAIL_NOT_WRITABLE}} |
+| 126 | install-update: update | Command `/bin/magento setup:upgrade` failed | {{site.data.cloud-error-messages.INSTALL_UPGRADE_ACTION}} |
+| 127 | clean-cache | Command `/bin/magento cache:flush` failed | {{site.data.cloud-error-messages.CLOUD_LOG_VERBOSE_ACTION}} |
+| 128 | disable-maintenance-mode | Command `/bin/magento maintenance:disable` failed | |
```
Comment (oshmyheliuk) [inline src/cloud/trouble/error-codes.md:79]:
```suggestion
| 128 | disable-maintenance-mode | Command `/bin/magento maintenance:disable` failed | {{site.data.cloud-error-messages.CLOUD_LOG_VERBOSE_ACTION}} |
```

File: src/_data/cloud-error-messages.yml [hunk 7]
```diff
@@ -0,0 +1,86 @@
+# Error messages
```
Comment (hguthrie) [inline src/_data/cloud-error-messages.yml:1]: The PR didn't explain, but is this file generated or copied?

Comment (meker12) [reply src/_data/cloud-error-messages.yml:1]: The Suggested actions are generated from a YAML file sourced from info provided by devs.  Devs are planning to create the YAML for the error messages in the next update (after this ece-tools release goes out).

File: src/cloud/trouble/error-codes.md (hunk 6 above)
Comment (meker12) [reply src/cloud/trouble/error-codes.md]: Added the missing suggested action locally and pushed to the PR.

File: src/cloud/trouble/error-codes.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
+---
+group: cloud-guide
+title: Error message reference for ece-tools
+---
+
+{% assign errors = site.data.cloud-error-messages %}
+
+This {{site.data.var.ct}} error message reference provides information to resolve errors that occur during the {{site.data.var.ece}} deployment process.
+
+Error messages are categorized by deployment stage–build, deploy, and post-deploy. Each section provides a list of associated errors with the following information for each error:
```
Comment (hguthrie) [inline src/cloud/trouble/error-codes.md:10]:
```suggestion
Error messages are categorized by deployment stage: build, deploy, and post-deploy. Each section provides a list of associated errors with the following information for each error:
```

Also, you start this paragraph about the deployment steps, but then context switch to "Sections", which I assumes means the tables below. I would clarify or change section to stage.

File: src/cloud/trouble/error-codes.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+{: .error-table}
+| Error code | Build step | Error description | Suggested action |
+| ---------- | ---------- | ----------------- | ---------------- |
+| 2 | | Cannot write to the `./app/etc/env.php` file | {{site.data.cloud-error-messages.ENV_PHP_IS_NOT_WRITABLE}}|
+| 3 | | Configuration isn't defined in the `schema.yaml` file | {{site.data.cloud-error-messages.CONFIG_NOT_DEFINED}}|
+| 4 | | Failed to parse the `.magento.env.yaml` file | {{site.data.cloud-error-messages.CONFIG_PARSE_FAILED}}|
+| 5 | | Unable to read the `.magento.env.yaml` file | {{site.data.cloud-error-messages.CONFIG_UNABLE_TO_READ}}|
+| 6 | | Unable to read the `.schema.yaml` file | |
+| 7 | refresh-modules | Cannot write to the `./app/etc/config.php` file |{{site.data.cloud-error-messages.CONFIG_PHP_IS_NOT_WRITABLE}} |
```
Comment (hguthrie) [inline src/cloud/trouble/error-codes.md:27]: I do like the tables. It really helps with self-sufficiency when you have reference material like this. I do think the Build step is easier to see if it is formatted as inline code. Give it a try and see what you think.

```suggestion
| 7 | `refresh-modules` | Cannot write to the `./app/etc/config.php` file |{{site.data.cloud-error-messages.CONFIG_PHP_IS_NOT_WRITABLE}} |
```

Comment (meker12) [reply src/cloud/trouble/error-codes.md:27]: @hguthrie  Devs added some additional updates. I'll add your suggestions in another PR. Thanks for the review.

**Pull request merged**